File: AdventOfCode/2023/01/python/day01_part01_retrieve_calibration_value.py
import sys

target_file = "../input.txt" if len(sys.argv) < 2 else sys.argv[1]

with open(target_file, 'r') as file_handler:
    lines = file_handler.readlines()

sum_all_calibration_values = 0

for line in lines:
    # Find the first and last digit in the line
    first_digit = None
    last_digit = None
    for s in line:
        if s.isdigit():
            if first_digit is None:
                first_digit = int(s)
            last_digit = int(s)
    sum_all_calibration_values += first_digit*10 + last_digit

print("Sum of all calibration values is", sum_all_calibration_values)

File: AdventOfCode/2023/01/js/day01_part01_retrieve_calibration_value.js
const fs = require('fs');
const process = require('process');

// Set the target file; use "../input.txt" by default or take from command line argument
let targetFile = process.argv.length < 3 ? "../input.txt" : process.argv[2];

function readFile(callback) {
    fs.readFile(targetFile, 'utf8', callback);
}

function processFile(err, data) {
    if (err) {
        console.error("Error reading the file:", err);
        return;
    }

    let lines = data.split('\n');
    let sumAllCalibrationValues = 0;

    for (let line of lines) {
        // Find the first and last digit in the line
        let firstDigit = null;
        let lastDigit = null;
        for (let s of line) {
            if (!isNaN(parseInt(s))) {
                lastDigit = parseInt(s);
                if (firstDigit === null) {
                    firstDigit = lastDigit;
                }
            }
        }

        if (firstDigit !== null) {
            sumAllCalibrationValues += firstDigit * 10 + lastDigit;
        }
    }

    console.log("Sum of all calibration values is", sumAllCalibrationValues);
}

readFile(processFile);
